dspace-api/src/main/java/org/dspace/content/DSpaceObject.java

```diff
@@ -192,4 +202,70 @@ public void clearModified() {
     protected void setModified() {
         this.modified = true;
     }
+
+    public String toString() {
+        String ident = getHandle();
+        if (ident == null)
+            ident = "" + getID();
+        return Constants.typeText[getType()] + "." + ident;
+    }
+
+    /**
+     * expects a handle or a two part string as parameter: TYPE.IDENT,
+     * where TYPE is a DSpaceObject type and
+     * IDENT is a UUID, handle,  group name, or Eperson's netid
+     *
+     * returns DSpaceObject that corresponds to string or null
+     * @param str    must have format TYPE.ID
+     * @return
+     */
+    public static DSpaceObject fromString(Context c, String str) throws SQLException {
```

Comment:
not sure how this would  work - fromString  returns an Item, Collection, Group, Eperson or whatever object type is encoded in its string parameter;  but in the content service factory   I can only ask for an  DSpaceObjectService for a particular type; 

I could of cause add the  fromString method to DSpaceObjectService and then use it through any of the implementing classes; eg  itemService.fromString("EPERSON.monikam")   - which would behaves oddly since it returns an Eperson not an Item 

fromString is heavily used to parse CLI arguments into the respective objects